preview/MsixCore/msixmgr/CryptoProvider.cpp

```diff
@@ -0,0 +1,191 @@
+#include <Windows.h>
+#include <WinCrypt.h>
+
+#include <BcryptLibrary.hpp>
+#include "GeneralUtil.hpp"
+#include <CryptoProvider.hpp>
+#include <TraceLoggingProvider.h>
+#include "MsixTraceLoggingProvider.hpp"
+#include <assert.h>
+
+namespace MsixCoreLib
+{
+    CryptoProvider::CryptoProvider()
+    {
+        this->providerHandle = NULL;
+        this->hashHandle = NULL;
+        this->digest.bytes = NULL;
+        this->digest.length = 0;
+    }
+
+    CryptoProvider::~CryptoProvider()
+    {
+        Reset();
+    }
+
+    void CryptoProvider::Reset()
+    {
+        this->digest.bytes = NULL;
+        this->digest.length = 0;
+
+        if (NULL != this->hashHandle)
+        {
+            if (!SUCCEEDED(BcryptLibrary::BCryptDestroyHash(this->hashHandle)))
+            {
+                return;
+            }
+            this->hashHandle = NULL;
+        }
+
+        if (NULL != this->providerHandle)
+        {
+            if (!SUCCEEDED(BcryptLibrary::BCryptCloseAlgorithmProvider(this->providerHandle, 0)))
+            {
+                return;
+            }
+            this->providerHandle = NULL;
+        }
+    }
+
+    HRESULT CryptoProvider::Create(
+        _Outptr_ CryptoProvider** provider)
+    {
+        RETURN_IF_FAILED(BcryptLibrary::Load());
+
+        std::unique_ptr<CryptoProvider> cp(new APPXCOMMON_NEW_TAG CryptoProvider());
+        //RETURN_IF_NULL_ALLOC(cp);
+
+        *provider = cp.release();
+        return S_OK;
+    }
+
+    HRESULT CryptoProvider::OpenProvider()
+    {
+        assert(NULL == this->providerHandle);
+
+        PCWSTR algorithmId = BCRYPT_SHA256_ALGORITHM;
+        PCWSTR implementation = NULL;
+        ULONG flags = 0;
+
+        BcryptLibrary::BCryptOpenAlgorithmProvider(
+            &this->providerHandle,
+            algorithmId,
+            implementation,
+            flags);
+
+        return S_OK;
+    }
+
+    HRESULT CryptoProvider::StartDigest()
+    {
+        BYTE* hashObject;
+        ULONG hashObjectSize;
+        ULONG resultSize;
+
+        assert(reset == this->state);

```

Comment:
>assert [](start = 8, length = 6)

get rid of these

Reply:
Removed all asserts for CryptoProvider.cpp.